### PR TITLE
Refactored some logging statements to use lazy % formatting

### PR DESCRIPTION
**Pull request by WhiteWolf47**
Refactored some logging statements in [bugbug/scripts/regressor_finder.py](https://github.com/mozilla/bugbug/blob/69972a1684f788319bf5c2944bbe8eeb79428c7d/scripts/regressor_finder.py#L355) 
to use % lazy format.

This PR is part of the fix for #3323 .
This goal of this PR is to refactor some of the logging statements in [bugbug/scripts/regressor_finder.py](https://github.com/mozilla/bugbug/blob/69972a1684f788319bf5c2944bbe8eeb79428c7d/scripts/regressor_finder.py#L355)  to use the % lazy style of formatting
for example : refactoring this statement   ->    logger.info("Analyzing {}...".format(bug_fixing_commit["rev"])) 
     to     logger.info("Analyzing %s...", bug_fixing_commit["rev"])

### CODE REVIEW

**Comment by WhiteWolf47** [comment]
@suhaibmujahid can you please review this? :blush:

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,6 @@ node_modules/
 *.log
 # Desktop Service Store
 *.DS_Store
+
+#Python virtual environment
+bb_bug/
```
Comment (suhaibmujahid) [inline .gitignore:55]: Could you please drop unrelated changes?

**Review by suhaibmujahid — CHANGES_REQUESTED** [review]
The goal of #3323 is to refactor all cases. However, you do not need to refactor all of them yourself. Will close #3323 when there are no more cases left.

BTW. This pull request is not linked to an issue. See here how to link a pull request to an issue: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: .gitignore (hunk 1 above)
Comment (WhiteWolf47) [reply .gitignore:55]: Okay I'll drop the unrelated changes

Comment (suhaibmujahid) [reply .gitignore:55]: Thank you, @WhiteWolf47!

Comment (WhiteWolf47) [reply .gitignore]: @suhaibmujahid is this alright now?

File: .gitignore [hunk 2]
```diff
@@ -49,4 +49,4 @@ node_modules/
 # Logs
 *.log
 # Desktop Service Store
-*.DS_Store
+*.DS_Store
```
Comment (suhaibmujahid) [inline .gitignore:52]: Unrelated :)

Comment (WhiteWolf47) [reply .gitignore:52]: my bad :)

Comment (WhiteWolf47) [reply .gitignore:52]: but aint that the same thing?

Comment (WhiteWolf47) [reply .gitignore:52]: i created a python virtual env but then i deleted it and removed it from the gitignore file, everything else is the same. I dont know why its showing that but that shouldnt be a problem .

**Comment by suhaibmujahid** [comment]
@WhiteWolf47 I think resolving #3323 is not your goal.  Could you please update the description to mention that this pull request is part of the fix for #3323? Also, could you reflect the goal of the pull request in the description? The description mentions `regressor_finder.py` where the patch is changing `regressor_finder.py` as well.

**Comment by WhiteWolf47** [comment]
> @WhiteWolf47 I think resolving #3323 is not your goal. Could you please update the description to mention that this pull request is part of the fix for #3323? Also, could you reflect the goal of the pull request in the description? The description mentions `regressor_finder.py` where the patch is changing `regressor_finder.py` as well.

Will do that!

File: .gitignore (hunk 2 above)
Comment (suhaibmujahid) [reply .gitignore:52]: It is about the new line at the end of the file. Your patch removed it.

File: bugbug/models/duplicate.py [hunk 3]
```diff
@@ -147,4 +147,4 @@ def get_labels(self):
         return classes, [0, 1]
 
     def get_feature_names(self):
-        return self.extraction_pipeline.named_steps["union"].get_feature_names_out()
+        return self.extraction_pipeline.named_steps["union"].get_feature_names_out()
```
Comment (suhaibmujahid) [inline bugbug/models/duplicate.py:150]: Unrelated change. Similar to the unrelated change in `.gitignore`.

**Review by suhaibmujahid — CHANGES_REQUESTED** [review]
Could you please discard unrelated cahnges (the deleted lines at the end of `.gitignore` and `scripts/regressor_finder.py`)?

**Comment by suhaibmujahid** [comment]
Also, please fix the lint errors. 

To avoid lint failures, it is recommended to setup `pre-commit`: https://github.com/mozilla/bugbug#auto-formatting

File: bugbug/models/duplicate.py (hunk 3 above)
Comment (WhiteWolf47) [reply bugbug/models/duplicate.py:150]: You mean that I should add the empty line at the end of those files back?

Comment (suhaibmujahid) [reply bugbug/models/duplicate.py:150]: Yes

**Comment by WhiteWolf47** [comment]
@suhaibmujahid i've setup the pre-commit , still the lint error is there :thinking:

**Comment by suhaibmujahid** [comment]
> @suhaibmujahid i've setup the pre-commit , still the lint error is there 🤔 . Also the pre-commit tests are passing perfectly normal in my local repo

pre-commit runs on the new changes only. If the change is already committed, it will ignore it, but the CI will fail. To run pre-commit on everything, please run `pre-commit run --all-files`.

**Comment by WhiteWolf47** [comment]
> > @suhaibmujahid i've setup the pre-commit , still the lint error is there thinking . Also the pre-commit tests are passing perfectly normal in my local repo
> 
> pre-commit runs on the new changes only. If the change is already committed, it will ignore it, but the CI will fail. To run pre-commit on everything, please run `pre-commit run --all-files`.

okay lemme try this!

**Comment by WhiteWolf47** [comment]
@suhaibmujahid "hook id : black" this one has failed, everything else has passed

File: scripts/regressor_finder.py [hunk 4]
```diff
@@ -64,15 +64,18 @@ def __init__(
...
                 )
 
             if self.tokenized_git_repo_url is not None:
                 logger.info(
-                    f"Cloning {self.tokenized_git_repo_url} to {self.tokenized_git_repo_dir}..."
+                    "Cloning %s to %s...",
+                    self.tokenized_git_repo_url,
+                    self.tokenized_git_repo_dir,
                 )
+
```
Comment (suhaibmujahid) [inline scripts/regressor_finder.py:78]:
```suggestion
```
Nit: Could toy remove this new line?

**Review by suhaibmujahid — APPROVED** [review]
LGTM. Thank you, @WhiteWolf47!

File: scripts/regressor_finder.py (hunk 4 above)
Comment (WhiteWolf47) [reply scripts/regressor_finder.py]: done

**Comment by WhiteWolf47** [comment]
> LGTM. Thank you, @WhiteWolf47!

No problem man. Thanks for helping me out with the issue. Looking forward to more contributions to bugbug :smiley:

**Comment by WhiteWolf47** [comment]
@suhaibmujahid did the recommended changes, can you please review and merge now :blush:

**Review by suhaibmujahid — APPROVED** [review]
Thank you!
